packages/commons/src/interop-token/models.ts

```diff
@@ -3,134 +3,255 @@ import {
-  client_id: ClientId;
-  sub: ClientId;
-  [ORGANIZATION_ID_CLAIM]: TenantId;
-};
+// ==========================================
+//     Interop M2M Token
+// ==========================================
+export const InteropJwtApiCommonPayload = InteropJwtCommonPayload.merge(
+  z.object({
+    client_id: ClientId,
+    sub: ClientId,
+    organizationId: TenantId,
+  })
+);
+export type InteropJwtApiCommonPayload = z.infer<
+  typeof InteropJwtApiCommonPayload
+>;
 
-export type InteropJwtApiM2MPayload = InteropJwtApiCommonPayload & {
-  [ROLE_CLAIM]: Extract<SystemRole, "m2m">;
-};
+export const InteropJwtApiM2MPayload = InteropJwtApiCommonPayload.merge(
+  z.object({ role: z.literal(systemRole.M2M_ROLE) })
+);
+export type InteropJwtApiM2MPayload = z.infer<typeof InteropJwtApiM2MPayload>;
 
-export type InteropJwtApiM2MAdminPayload = InteropJwtApiCommonPayload & {
-  [ROLE_CLAIM]: Extract<SystemRole, "m2m-admin">;
-  adminId: UserId;
-  // ^ ID of the admin user associated with the client
-};
+// ==========================================
+//   Interop M2M Admin Token
+// ==========================================
+export const InteropJwtApiM2MAdminPayload = InteropJwtApiCommonPayload.merge(
+  z.object({
+    role: z.literal(systemRole.M2M_ADMIN_ROLE),
+    adminId: z.string().uuid(),
+    // ^ ID of the admin user associated with the client
+  })
+);
+export type InteropJwtApiM2MAdminPayload = z.infer<
+  typeof InteropJwtApiM2MAdminPayload
+>;
 
+// ==========================================
+//   Interop API Token
+// ==========================================
 export type InteropJwtApiPayload =
   | InteropJwtApiM2MAdminPayload
   | InteropJwtApiM2MPayload;
 
+/* ========================================== 
+    Interop API Token 
+  ========================================== */
+
 export type InteropApiToken = {
   header: InteropJwtHeader;
   payload: InteropJwtApiPayload;
   serialized: string;
 };
 
-/* ========================================== 
-    Interop INTERNAL Token 
-  ========================================== */
-export type InteropJwtInternalPayload = InteropJwtCommonPayload & {
-  sub: string;
-  role: Extract<SystemRole, "internal">;
-};
+// ==========================================
+//   Interop INTERNAL Token
+// ==========================================
+export const InternalJwtAuthPayload = InteropJwtCommonPayload.merge(
+  z.object({
+    role: z.literal(systemRole.INTERNAL_ROLE),
+    sub: z.string(),
+  })
+);
+export type InternalJwtAuthPayload = z.infer<typeof InternalJwtAuthPayload>;
+
+// ==========================================
+//  Interop MAINTENANCE Token
+// ==========================================
+export const InteropJwtMaintenancePayload = InteropJwtCommonPayload.merge(
+  z.object({
+    role: z.literal(systemRole.MAINTENANCE_ROLE),
+    sub: z.string(),
+  })
+);
+export type InteropJwtMaintenancePayload = z.infer<
+  typeof InteropJwtMaintenancePayload
+>;
+
```

Comment:
In the previous sections, you also defined the token, not only the payload (where you defined `InteropConsumerToken`, `InteropApiToken`). Why do you define only the payload here? Is it because we don't generate Maintenance tokens, so the payload is enough since we need it only for parsing?